geoopt/samplers/sgld.py

```diff
@@ -0,0 +1,86 @@
+import math
+import numpy as np
+import torch
+
+from ..tensor import ManifoldParameter, ManifoldTensor
+from ..manifolds import Rn
+from ..optim.mixin import OptimMixin
+from .base import Sampler
+
+
+class SGLD(Sampler):
+    """Stochastic Gradient Langevin Dynamics"""
+
+    def __init__(self, params, epsilon=1e-3):
+        defaults = dict(epsilon=epsilon)
+        super().__init__(params, defaults)
+    
+    
+    def step(self, closure):
+        """Performs a single sampling step.
+
+        Arguments:
+            closure (callable, optional): A closure that reevaluates the model
+                and returns the log probability.
+        """
+        logp = closure()
+        logp.backward()
+
+        with torch.no_grad():
+            for group in self.param_groups:
+                for p in group['params']:
+                    epsilon = group['epsilon']
+
+                    n = torch.randn_like(p).mul_(math.sqrt(epsilon))
+                    r = 0.5 * epsilon * p.grad + n
+
+                    p.data.add_(r)
+                    p.grad.data.zero_()
+
+        if not self.burnin:
+            self.steps += 1
+            self.log_probs.append(logp.item())
+
+
+class RSGLD(OptimMixin, SGLD):
```

Comment:
Why not using just one class? It seems to me that code is too duplicated. Riemannian version should work for trivial manifold as well. I know, there are some performance issues with additional copying, but they could be resolved easily with refactoring after the PR